stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -192,8 +190,10 @@ public class ChatClient internal constructor(
             }
         }
         logger.logI("Initialised: " + getVersion())
+    }
 
-        plugins.forEach { it.init(appContext, this) }
+    internal fun addPlugins(plugins: List<Plugin>) {
+        this.plugins = plugins
```

Comment:
It would be nice to have `plugins` in the constructor, but OfflinePlugin needs an initialization logic which must happen after `ChatClient` initialization. Then plugins are added by the builder after `ChatClient` is initialized.

Reply:
I would prefer to avoid this `addPlugins` as well.
If we have the `PluginFactory`, why don't we use it to create plugins when they are needed? With it we will have `chatClient` already instanciated and be able to inject into `OfflinePlugin`.
Am I missing anything? :thinking:

Reply:
@JcMinarro. If initialize then only when we need it, then it also won't go in the constructor. We can have the factories in the constructor and initialize the plugins when they are needed. I didn't want to expose the factories in the constructor of ChatClient, but this could be a way to initialize the plugins lazilly. But let me see if I can remove ChatDomain from OfflinePlugin what happens then.

Reply:
So... Removing `ChatDomain` from `StateRegistry` is easy, but to remove it from `StreamOfflinePluginFactory.createOfflinePlugin` is much harder because much of the initialization of the SDK happens inside `ChatDomain`. 

We need to remove the initialization of `RepositoryFacade (this shouldn't be there anyway...), a way to provide the scopes and place to put and provide `lastUsers`. Let's talk about it over chat, I prefer this as a separated task.